### PR TITLE
Fix/clean up various ibus-engines

### LINKED ISSUE #68837: ibus with libpinyin stopped working after upgrade to nixos-19.09 channel
**Describe the bug**
I configured https://nixos.org/channels/nixos-19.09 to be my new nixos channel and ran `nixos-rebuild switch` followed by a manual reboot.

My ibus was configured so that I can switch between three input methods (“Chinese - Bopomofo”, “Chinese - Intelligent Pinyin”, and “German - German” with <Super>y / Windows-Y.

The Windows-Y key combination doesn't seem to have an effect anymore. Nothing gets changed, when I press the Windows key and Y, no small window indicating the selected input method gets shows (this was done before the change to 19.09).

Here are the relevant parts of my `configuration.nix`:

```nix
{
  i18n = {
    consoleFont = "Lat2-Terminus16";
    consoleKeyMap = "de";
    defaultLocale = "zh_TW.UTF-8";
    inputMethod = {
      enabled = "ibus";
      ibus.engines = with pkgs.ibus-engines; [ libpinyin mozc ];
    };
  };

  environment.systemPackages = with pkgs; [
    wget vim lvm2 cryptsetup emacs firefox mutt git bash-completion dmenu xorg.xmodmap xorg.libXinerama rxvt_unicode ibus ibus-engines.libpinyin libpinyin ibus-engines.mozc xorg.xev file anki libreoffice-fresh unzip evince patchelf gnome3.gnome-terminal feh nixops usbutils w3m html-tidy haskellPackages.stack leiningen mercurial clojure openjdk gnumake aspell aspellDicts.de slack gimp gnuplot whois gcc dia xournal killall tree cmake binutils-unwrapped openssl gcc-arm-embedded-7 rustup gnupg gnome3.vte manpages pmutils numlockx udev signal-desktop nomacs x2goclient drone-cli pwgen gdb bind.dnsutils haskellPackages.xmobar haskellPackages.xmonad haskellPackages.xmonad-contrib haskellPackages.xmonad-extras eagle exercism clang-tools hlint pavucontrol weechat yubioath-desktop yubikey-manager yubikey-manager-qt yubikey-personalization yubikey-personalization-gui gnucash libchipcard aqbanking gwenhywfar
    (texlive.combine {
      inherit (texlive)
      scheme-medium a4wide bclogo changebar csquotes ctex dinbrief enumitem environ fandol fourier framed mdframed multirow needspace noto todonotes trimspaces xecjk xpinyin zhnumber wrapfig capt-of;
    })
  ];
}
```

When I run `ibus-setup` and select one of the input methods then pressing the Preferences button, nothing opens but I get the following message on the terminal window from which I started `ibus-setup`:

```
Traceback (most recent call last):
  File "main2.py", line 32, in <module>
    from gi import require_version as gi_require_version
ModuleNotFoundError: No module named 'gi'
```

**Expected behavior**
I would expect to be able to switch keyboard layouts as before with the Windows-Y key combination.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.2.13, NixOS, 19.09beta181.64e38f246ae (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(root): `"nixos-19.09beta181.64e38f246ae, unstable-19.09pre192418.e19054ab3cd"`
 - channels(matthias): `"home-manager-19.03"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
attribute:
- ibus
- ibus-engines.libpinyin
- ibus-engines.mozc
module:
- i18n.inputMethod
```

### PR DESCRIPTION
**Pull request by jtojnar**
Closes: https://github.com/NixOS/nixpkgs/issues/68837

### CODE REVIEW

**Comment by laMudri** [comment]
Are these changes relevant to any other IBus input methods?

**Comment by jtojnar** [comment]
Yes, all input methods should be checked and updated. I already did this in typing booster and some other in the past. Need to run now, will try to continue during the week.

Also did not test this other than by running the setup execs on non-ibus system.

**Comment by jtojnar** [comment]
The installed tests seem to be failing at the moment:

```
machine: must succeed: gnome-desktop-testing-runner  -d '/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share'
machine #
machine # (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-engine-switch:1200): IBUS-CRITICAL **: 22:23:28.055: ibus_bus_call_sync: assertion 'ibus_bus_is_connected (bus)' failed
machine # [   20.040629] do_trap: 1 callbacks suppressed
machine # [   20.040632] traps: ibus-engine-swi[1200] trap int3 ip:7fac2e81dc75 sp:7ffcabaebfc0 error:0 in libglib-2.0.so.0.6200.1[7fac2e7e4000+80000]
machine # [   19.949938] systemd[1]: Created slice system-systemd\x2dcoredump.slice.
machine # [   19.961307] systemd[1]: Started Process Core Dump (PID 1202/UID 0).
machine # [   20.541447] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-engine-switch.test (Child process killed by signal 5)
machine # [   20.566825] systemd-coredump[1203]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   20.573043] systemd-coredump[1203]: Process 1200 (ibus-engine-swi) of user 0 dumped core.
machine # [   20.593256] systemd[1]: systemd-coredump@0-1202-0.service: Succeeded.
machine # (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-inputcontext-create:1205): GLib-GIO-CRITICAL **: 22:23:28.748: g_dbus_connection_call_internal: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
machine #
machine # [   20.734501] traps: ibus-inputconte[1205] trap int3 ip:7ff194fdfc75 sp:7ffe10edcdd0 error:0 in libglib-2.0.so.0.6200.1[7ff194fa6000+80000]
machine # [   20.646789] systemd[1]: Started Process Core Dump (PID 1211/UID 0).
machine # [   21.067351] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-inputcontext-create.test (Child process killed by signal 5)
machine # [   21.089903] systemd-coredump[1212]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   21.095815] systemd-coredump[1212]: Process 1205 (ibus-inputconte) of user 0 dumped core.
machine # [   21.119609] systemd[1]: systemd-coredump@1-1211-0.service: Succeeded.
machine #
machine # (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-factory:1214): IBUS-CRITICAL **: 22:23:29.292: ibus_factory_new: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
machine # [   21.279024] traps: ibus-factory[1214] trap int3 ip:7fb187562c75 sp:7fff009d63f0 error:0 in libglib-2.0.so.0.6200.1[7fb187529000+80000]
machine # [   21.203275] systemd[1]: Started Process Core Dump (PID 1221/UID 0).
machine # [   21.615804] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-factory.test (Child process killed by signal 5)
machine # [   21.641508] systemd-coredump[1222]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   21.648878] systemd-coredump[1222]: Process 1214 (ibus-factory) of user 0 dumped core.
machine #
machine # (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-configservice:1224): IBUS-CRITICAL **: 22:23:29.820: ibus_config_service_new: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
machine # [   21.805027] traps: ibus-configserv[1224] trap int3 ip:7fb8a2223c75 sp:7ffcbe916da0 error:0 in libglib-2.0.so.0.6200.1[7fb8a21ea000+80000]
machine # [   21.674715] systemd[1]: systemd-coredump@2-1221-0.service: Succeeded.
machine # [   21.716889] systemd[1]: Started Process Core Dump (PID 1227/UID 0).
machine # [   22.160533] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-configservice.test (Child process killed by signal 5)
machine # [   22.186699] systemd-coredump[1231]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   22.192996] systemd-coredump[1231]: Process 1224 (ibus-configserv) of user 0 dumped core.
machine # [   22.211027] systemd[1]: systemd-coredump@3-1227-0.service: Succeeded.
machine #
machine # ** (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-bus:1233): WARNING **: 22:23:30.375: Not connected to ibus-daemon
machine # [   22.360749] traps: ibus-bus[1233] trap int3 ip:7f24f18a4c75 sp:7ffc54d9ceb0 error:0 in libglib-2.0.so.0.6200.1[7f24f186b000+80000]
machine # [   22.287361] systemd[1]: Started Process Core Dump (PID 1239/UID 0).
machine # [   22.801921] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-bus.test (Child process killed by signal 5)
machine # [   22.833272] systemd-coredump[1240]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   22.839758] systemd-coredump[1240]: Process 1233 (ibus-bus) of user 0 dumped core.
machine # [   22.862416] systemd[1]: systemd-coredump@4-1239-0.service: Succeeded.
machine # [   22.877867] gnome-desktop-testing-runner[1198]: PASS: flatpak/ibus-keynames.test
machine # **
machine # IBUS:ERROR:ibusconfig.c:333:ibus_config_new_async: assertion failed: (G_IS_DBUS_CONNECTION (connection))
machine # [   22.979814] systemd[1]: Started Process Core Dump (PID 1250/UID 0).
machine # [   23.398322] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-config.test (Child process killed by signal 6)
machine # [   23.422856] systemd-coredump[1251]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   23.430031] systemd-coredump[1251]: Process 1248 (ibus-config) of user 0 dumped core.
machine # [   23.444975] systemd[1]: systemd-coredump@5-1250-0.service: Succeeded.
machine # [   23.457119] gnome-desktop-testing-runner[1198]: PASS: flatpak/ibus-share.test
machine # [   23.747784] gnome-desktop-testing-runner[1198]: PASS: flatpak/ibus-util.test
machine #
machine # (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-inputcontext:1261): IBUS-CRITICAL **: 22:23:31.933: ibus_bus_call_sync: assertion 'ibus_bus_is_connected (bus)' failed
machine # [   23.917626] traps: ibus-inputconte[1261] trap int3 ip:7ffab6a32c75 sp:7ffc73facc50 error:0 in libglib-2.0.so.0.6200.1[7ffab69f9000+80000]
machine # [   23.819277] systemd[1]: Started Process Core Dump (PID 1263/UID 0).
machine # [   24.215727] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-inputcontext.test (Child process killed by signal 5)
machine # [   24.241155] systemd-coredump[1264]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   24.246436] systemd-coredump[1264]: Process 1261 (ibus-inputconte) of user 0 dumped core.
machine # [   24.264670] systemd[1]: systemd-coredump@6-1263-0.service: Succeeded.
machine # [   24.275318] gnome-desktop-testing-runner[1198]: PASS: flatpak/ibus-serializable.test
machine # [   24.308327] gnome-desktop-testing-runner[1198]: PASS: flatpak/ibus-registry.test
machine #
machine # ** (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose:1280): CRITICAL **: 22:23:33.103: ibus-daemon is not running.
machine # [   25.087208] traps: ibus-compose[1280] trap int3 ip:7fde98839c75 sp:7ffd15501ce0 error:0 in libglib-2.0.so.0.6200.1[7fde98800000+80000]
machine # [   24.990614] systemd[1]: Started Process Core Dump (PID 1282/UID 0).
machine # [   25.001409] gnome-desktop-testing-runner[1198]: Executing: flatpak/ibus-compose-locales.test
machine # /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose-locales: line 12:  1280 Trace/breakpoint trap   (core dumped) env $var $BUILDDIR/ibus-compose $SRCDIR $@
machine # ls: cannot access '*.cache': No such file or directory
machine # [   25.581960] systemd-coredump[1283]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   25.586941] systemd-coredump[1283]: Process 1280 (ibus-compose) of user 0 dumped core.
machine # [   25.598021] systemd[1]: systemd-coredump@7-1282-0.service: Succeeded.
machine #
machine # ** (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose:1289): CRITICAL **: 22:23:33.918: ibus-daemon is not running.
machine # [   25.902847] traps: ibus-compose[1289] trap int3 ip:7fc816694c75 sp:7ffd73570040 error:0 in libglib-2.0.so.0.6200.1[7fc81665b000+80000]
machine # [   25.800258] systemd[1]: Started Process Core Dump (PID 1295/UID 0).
machine # /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose-locales: line 12:  1289 Trace/breakpoint trap   (core dumped) env $var $BUILDDIR/ibus-compose $SRCDIR $@
machine # [   26.245523] systemd-coredump[1296]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   26.249404] systemd-coredump[1296]: Process 1289 (ibus-compose) of user 0 dumped core.
machine # ls: cannot access '*.cache': No such file or directory
machine # [   26.268536] systemd[1]: systemd-coredump@8-1295-0.service: Succeeded.
machine #
machine # ** (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose:1306): CRITICAL **: 22:23:34.580: ibus-daemon is not running.
machine # [   26.564222] traps: ibus-compose[1306] trap int3 ip:7fb5da04fc75 sp:7ffd16aa8690 error:0 in libglib-2.0.so.0.6200.1[7fb5da016000+80000]
machine # [   26.472804] systemd[1]: Started Process Core Dump (PID 1308/UID 0).
machine # /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose-locales: line 12:  1306 Trace/breakpoint trap   (core dumped) env $var $BUILDDIR/ibus-compose $SRCDIR $@
machine # [   26.907743] systemd-coredump[1309]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   26.912778] systemd-coredump[1309]: Process 1306 (ibus-compose) of user 0 dumped core.
machine # ls: cannot access '*.cache': No such file or directory
machine # [   26.933691] systemd[1]: systemd-coredump@9-1308-0.service: Succeeded.
machine #
machine # ** (/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose:1319): CRITICAL **: 22:23:35.209: ibus-daemon is not running.
machine # [   27.193324] traps: ibus-compose[1319] trap int3 ip:7f96c1f49c75 sp:7ffe0f9f6620 error:0 in libglib-2.0.so.0.6200.1[7f96c1f10000+80000]
machine # [   27.106139] systemd[1]: Started Process Core Dump (PID 1321/UID 0).
machine # /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose-locales: line 12:  1319 Trace/breakpoint trap   (core dumped) env $var $BUILDDIR/ibus-compose $SRCDIR $@
machine # [   27.535096] systemd-coredump[1322]: Cannot resolve systemd-coredump user. Proceeding to dump core as root: No such process
machine # [   27.540083] systemd-coredump[1322]: ls: cannot access '*.cache': No such file or directory
machine # Process 1319 (ibus-compose) of user 0 dumped core.
machine # [   27.553815] gnome-desktop-testing-runner[1198]: FAIL: flatpak/ibus-compose-locales.test (Child process exited with code 20)
machine # [   27.572398] systemd[1]: systemd-coredump@10-1321-0.service: Succeeded.
machine # [   27.577446] gnome-desktop-testing-runner[1198]: SUMMARY: total=13; passed=5; skipped=0; failed=8; user=0.5s; system=0.4s; maxrss=54760
machine: output: Running test: flatpak/ibus-engine-switch.test
# random seed: R02S48983ca3e31d2076482e63ac70c9e21f
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
1..4
# Start of ibus tests
# Start of engine-switch tests
Bail out! IBUS-FATAL-CRITICAL: ibus_bus_call_sync: assertion 'ibus_bus_is_connected (bus)' failed
Running test: flatpak/ibus-inputcontext-create.test
# random seed: R02S7f80edc34756d5fd348141a994fe490e
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
1..2
# Start of ibus tests
Bail out! GLib-GIO-FATAL-CRITICAL: g_dbus_connection_call_internal: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
Running test: flatpak/ibus-factory.test
# random seed: R02S7fe1f2f0d5ffe2d87404ac61f3e68f97
1..1
# Start of ibus tests
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
Bail out! IBUS-FATAL-CRITICAL: ibus_factory_new: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
Running test: flatpak/ibus-configservice.test
# random seed: R02S2e101f470a1410f62ca4c3cd86ad18b9
1..1
# Start of ibus tests
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
Bail out! IBUS-FATAL-CRITICAL: ibus_config_service_new: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
Running test: flatpak/ibus-bus.test
# random seed: R02S918d87077088248a49c918bc08ce45cc
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
Bail out! FATAL-WARNING: Not connected to ibus-daemon
Running test: flatpak/ibus-keynames.test
# random seed: R02S4a42875614860a300a6d3d9ab8d4e9b9
1..1
# Start of ibus tests
ok 1 /ibus/keyname
# End of ibus tests
Running test: flatpak/ibus-config.test
# random seed: R02S4f113a4cfec0d4e352e3dd4c35ddfe15
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
1..2
# Start of ibus tests
Bail out! IBUS:ERROR:ibusconfig.c:333:ibus_config_new_async: assertion failed: (G_IS_DBUS_CONNECTION (connection))
Running test: flatpak/ibus-share.test
# random seed: R02S3a3bf04c1b9129e030ed902f16d42a9f
1..1
# Start of ibus tests
ok 1 /ibus/marchine-id
# End of ibus tests
Running test: flatpak/ibus-util.test
Running test: flatpak/ibus-inputcontext.test
# random seed: R02S241d4a70d94862c083712dde6767f034
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
1..2
# Start of ibus tests
Bail out! IBUS-FATAL-CRITICAL: ibus_bus_call_sync: assertion 'ibus_bus_is_connected (bus)' failed
Running test: flatpak/ibus-serializable.test
# random seed: R02Sda75e97d9062dcb3358e0f9cc46550e5
1..7
# Start of ibus tests
ok 1 /ibus/varianttypeinfo
ok 2 /ibus/attrlist
ok 3 /ibus/text
ok 4 /ibus/enginedesc
ok 5 /ibus/lookuptable
ok 6 /ibus/property
ok 7 /ibus/attachment
# End of ibus tests
Running test: flatpak/ibus-registry.test
Running test: flatpak/ibus-compose-locales.test
# Starting LANG=el_GR.UTF-8 /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap
# random seed: R02S170ec158360c08b3163f0bbc36954791
1..1
# Start of ibus-compose tests
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
Bail out! FATAL-CRITICAL: ibus-daemon is not running.
# Finished LANG=el_GR.UTF-8 /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap with 133
# Starting LANG=fi_FI.UTF-8 /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap
# random seed: R02S3e1ec50f91dac743483da9f4248a7881
1..1
# Start of ibus-compose tests
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
Bail out! FATAL-CRITICAL: ibus-daemon is not running.
# Finished LANG=fi_FI.UTF-8 /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap with 266
# Starting LANG=pt_BR.UTF-8 /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap
# random seed: R02S976318110f1ac22c3e375e52e6606ab4
1..1
# Start of ibus-compose tests
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
Bail out! FATAL-CRITICAL: ibus-daemon is not running.
# Finished LANG=pt_BR.UTF-8 /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap with 399
# Starting LANG=en_US.UTF-8 COMPOSE_FILE=ibus-compose.emoji /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap
# random seed: R02S0b5b7437c90fd4cb2da64556dde2a91e
1..1
# Start of ibus-compose tests
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
Bail out! FATAL-CRITICAL: ibus-daemon is not running.
# Finished LANG=en_US.UTF-8 COMPOSE_FILE=ibus-compose.emoji /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/libexec/installed-tests/flatpak/ibus-compose /nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share/installed-tests/flatpak --tap with 532
FAIL: flatpak/ibus-engine-switch.test (Child process killed by signal 5)
FAIL: flatpak/ibus-inputcontext-create.test (Child process killed by signal 5)
FAIL: flatpak/ibus-factory.test (Child process killed by signal 5)
FAIL: flatpak/ibus-configservice.test (Child process killed by signal 5)
FAIL: flatpak/ibus-bus.test (Child process killed by signal 5)
FAIL: flatpak/ibus-config.test (Child process killed by signal 6)
FAIL: flatpak/ibus-inputcontext.test (Child process killed by signal 5)
FAIL: flatpak/ibus-compose-locales.test (Child process exited with code 20)

error: command `gnome-desktop-testing-runner  -d '/nix/store/h95r0bpnf6iwqdrqadvjcvl4zkf08w9r-ibus-1.5.21-installedTests/share'` failed (exit code 2)
```

I will need to check if adding `services.xserver.desktopManager.gnome3.enable = true;` to `testConfig` fixes that but that currently requires rebuild of rustc & webkitgtk.

**Comment by jtojnar** [comment]
Apparently, https://github.com/fujiwarat/ibus-anthy/commit/5a9e4858f1a2b6cad02ef1d272fc5bed959ec0bf is not yet merged to a stable release, so no tests for ibus-anthy.

Managed to get `nixosTests.installed-tests.ibus` down to a single failing test case.

**Comment by jtojnar** [comment]
I manually tested all the engines and they seem to work except for uniemoji and hangul.

Hopefully, the latest push fixes uniemoji but hangul does not seem to print any errors so more debugging will be required.

**Comment by jtojnar** [comment]
ibus-hangul 1.5.3 seems to have fixed the issue of it stopping:

https://github.com/libhangul/ibus-hangul/releases/tag/1.5.3

Now the newly added NixOS tests are still partially failing but all the input methods seem to work in VM, so I just marked the test as broken and we can fix that later.

**Comment by jtojnar** [comment]
The merging of the attrsets in f581284 is ugly but until we have https://github.com/NixOS/rfcs/pull/58, there is not much room for improvement.

File: pkgs/tools/inputmethods/ibus-engines/ibus-uniemoji/default.nix [hunk 1]
```diff
@@ -13,25 +24,40 @@ stdenv.mkDerivation rec {
...
+    # the wrapped script will have Python in shebang anyway.
+    ./allow-wrapping.patch
+  ];
 
-  nativeBuildInputs = [ python3Packages.wrapPython ];
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    gobject-introspection
+
```
Comment (worldofpeace) [inline pkgs/tools/inputmethods/ibus-engines/ibus-uniemoji/default.nix]: trail here

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Thanks, any test case would help with keeping up on these packages :+1: 
I went commit by commit, here's what I noticed.

* ibus-engines.libpinyin: 1.10.0 -> 1.11.1
Commit message needs indent fix.

* ibus-engines.anthy: clean up
I believe we should be using fetchFromGitHub for the fetch.
The commit msg has `* 1.5.11 switched from intltool to anthy`. probably meant gettext?

* ibus-engines.uniemoji: modernize
Let's use placeholder

* nixos/ibus: install D-Bus services
Not sure if I'm remembering this correctly, but could we get issues with multiple uses of
`programs.dconf.enable` in our modules?

**Comment by worldofpeace** [comment]
> The merging of the attrsets in f581284 is ugly but until we have [NixOS/rfcs#58](https://github.com/NixOS/rfcs/pull/58), there is not much room for improvement.

:+1: Yeah this RFC would make lots of code feel better.

**Comment by jtojnar** [comment]
>     * ibus-engines.libpinyin: 1.10.0 -> 1.11.1
>       Commit message needs indent fix.

The nested list is intentional as it highlights important changes in each release. There are not many, though.

>     * ibus-engines.anthy: clean up
>       I believe we should be using fetchFromGitHub for the fetch.

I keep confusing it too but `archive` is the VCS tarball, `releases/download` are the uploaded artefacts.

>       The commit msg has `* 1.5.11 switched from intltool to anthy`. probably meant gettext?

Oops, fixed.
 
>     * ibus-engines.uniemoji: modernize
>       Let's use placeholder

:+1:
 
>     * nixos/ibus: install D-Bus services
>       Not sure if I'm remembering this correctly, but could we get issues with multiple uses of
>       `programs.dconf.enable` in our modules?

It should not matter unless someone sets it to something different than `true`. But then it would fail at runtime, so the conflict would be fine.

**Review by worldofpeace — APPROVED** [review]
LGTM :+1: 
Favorite refactor was `pkgs/data/misc/unicode-emoji/default.nix` and `pkgs/tools/inputmethods/ibus/default.nix`.

**Pull request merged**